src/ophyd_async/epics/areadetector/drivers/ad_base.py

```diff
@@ -18,11 +37,29 @@ def __init__(self, prefix: str, name: str = "") -> None:
         self.array_counter = ad_rw(int, prefix + "ArrayCounter")
         self.array_size_x = ad_r(int, prefix + "ArraySizeX")
         self.array_size_y = ad_r(int, prefix + "ArraySizeY")
+        self.detector_state = ad_r(DetectorState, prefix + "DetectorState")
         # There is no _RBV for this one
         self.wait_for_plugins = epics_signal_rw(bool, prefix + "WaitForPlugins")
         super().__init__(prefix, name=name)
 
 
+async def arm_and_trigger_detector_and_check_status_pv(
+    driver: ADBase,
+    good_states: Set[DetectorState] = DEFAULT_GOOD_STATES,
+) -> AsyncStatus:
+    status = await set_and_wait_for_value(driver.acquire, True)
+
+    async def completion_task() -> None:
+        await status
```

Comment:
will this line not hang, meaning the rest of the logic never gets processed? 

I just remember that originally in the hdf writer wait_for_index it seems like we hung at await set_and_wait_for_value(...)